addon/lib/StudyTelemetryCollector.jsm

```diff
@@ -21,15 +21,88 @@ const BASERESOURCE = "esper-pioneer-shield-study";
 // XPCOMUtils.defineLazyModuleGetter(this, "studyUtils", STUDYUTILSPATH);
 */
 
+// telemetry utils
+const { TelemetrySession } = Cu.import("resource://gre/modules/TelemetrySession.jsm", null);
+
 /**
  * Note: Setting studyUtils in this constructor due to above comments
  */
 class StudyTelemetryCollector {
+
   constructor(studyUtils, variation) {
     this.variation = variation;
     this.studyUtils = studyUtils;
   }
 
   start() {
+
+    this.studyUtils.telemetry({ event: "esper-init" });
+
+    try {
+      this.collectAndSendTelemetry();
+    } catch (ex) {
+      // TODO: how are errors during study execution reported?
+      // this.studyUtils.telemetryError();
+      Components.utils.reportError(ex);
+    }
+
+  }
+
+  collectAndSendTelemetry() {
+
+    const shieldPingAttributes = this.collectShieldPingAttributes();
+
+    this.studyUtils.telemetry({
+      event: "telemetry-payload",
+      ...shieldPingAttributes,
```

Comment:
remember that all attributes must be strings, or it will throw validation errors.